diff --git a/CMakeLists.txt b/CMakeLists.txt
index f18cab0..3989c6a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,11 +65,12 @@ else()
   set(BOOST_COMPONENTS ${BOOST_COMPONENTS} signals)
 endif()
 
-find_package(Boost 1.46.0 REQUIRED COMPONENTS ${BOOST_COMPONENTS})
-if(Boost_FOUND)
-  include_directories(${Boost_INCLUDE_DIRS})
-  link_directories(${Boost_LIBRARY_DIRS})
-endif()
+#find_package(Boost 1.46.0 REQUIRED COMPONENTS ${BOOST_COMPONENTS})
+#if(Boost_FOUND)
+#  include_directories(${Boost_INCLUDE_DIRS})
+#  link_directories(${Boost_LIBRARY_DIRS})
+#endif()
+link_libraries(boost-mini)
 
 if(ENABLE_LOGGING)
 
diff --git a/thirdparty.mk b/thirdparty.mk
index 04c6f5a..bf9e167 100644
--- a/thirdparty.mk
+++ b/thirdparty.mk
@@ -25,6 +25,7 @@ clean-src:
 glog:
 	cd $(SRC_DIR)/glog; \
 	cmake . -Bcmake-build \
+	-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON \
 	-DBUILD_TESTING:BOOL=OFF \
 	-DWITH_GFLAGS:BOOL=OFF \
 	-DCMAKE_BUILD_TYPE:STRING="Release" \
@@ -34,6 +35,7 @@ glog:
 leveldb:
 	cd $(SRC_DIR)/leveldb; \
 	cmake . -Bbuild \
+	-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON \
 	-DLEVELDB_BUILD_BENCHMARKS:BOOL=OFF \
 	-DLEVELDB_BUILD_TESTS:BOOL=OFF \
 	-DCMAKE_BUILD_TYPE:STRING="Release" \
@@ -43,6 +45,7 @@ leveldb:
 marisa:
 	cd $(SRC_DIR)/marisa-trie; \
 	cmake $(SRC_DIR) -Bbuild \
+	-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON \
 	-DCMAKE_BUILD_TYPE:STRING="Release" \
 	-DCMAKE_INSTALL_PREFIX:PATH="$(THIRD_PARTY_DIR)" \
 	&& cmake --build build --target install
@@ -50,6 +53,7 @@ marisa:
 opencc:
 	cd $(SRC_DIR)/opencc; \
 	cmake . -Bbuild \
+	-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON \
 	-DBUILD_SHARED_LIBS:BOOL=OFF \
 	-DCMAKE_BUILD_TYPE:STRING="Release" \
 	-DCMAKE_INSTALL_PREFIX:PATH="$(THIRD_PARTY_DIR)" \
@@ -58,6 +62,7 @@ opencc:
 yaml-cpp:
 	cd $(SRC_DIR)/yaml-cpp; \
 	cmake . -Bbuild \
+	-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON \
 	-DYAML_CPP_BUILD_CONTRIB:BOOL=OFF \
 	-DYAML_CPP_BUILD_TESTS:BOOL=OFF \
 	-DYAML_CPP_BUILD_TOOLS:BOOL=OFF \
@@ -68,6 +73,7 @@ yaml-cpp:
 gtest:
 	cd $(SRC_DIR)/googletest; \
 	cmake . -Bbuild \
+	-DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON \
 	-DBUILD_GMOCK:BOOL=OFF \
 	-DCMAKE_BUILD_TYPE:STRING="Release" \
 	-DCMAKE_INSTALL_PREFIX:PATH="$(THIRD_PARTY_DIR)" \
