From 9d8b2e2eefb18373ee13bfb0733562ca76f15411 Mon Sep 17 00:00:00 2001
From: Chunhui He <hchunhui@mail.ustc.edu.cn>
Date: Sat, 13 Jul 2019 11:17:50 +0000
Subject: [PATCH 1/2] link boost mini

---
 CMakeLists.txt | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f18cab0..3989c6a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,11 +65,12 @@ else()
   set(BOOST_COMPONENTS ${BOOST_COMPONENTS} signals)
 endif()
 
-find_package(Boost 1.46.0 REQUIRED COMPONENTS ${BOOST_COMPONENTS})
-if(Boost_FOUND)
-  include_directories(${Boost_INCLUDE_DIRS})
-  link_directories(${Boost_LIBRARY_DIRS})
-endif()
+#find_package(Boost 1.46.0 REQUIRED COMPONENTS ${BOOST_COMPONENTS})
+#if(Boost_FOUND)
+#  include_directories(${Boost_INCLUDE_DIRS})
+#  link_directories(${Boost_LIBRARY_DIRS})
+#endif()
+link_libraries(boost-mini)
 
 if(ENABLE_LOGGING)
 
-- 
2.20.1

