From 894a679b1ffc30c20e213b58df25bc5197c8ed7b Mon Sep 17 00:00:00 2001
From: Chunhui He <hchunhui@mail.ustc.edu.cn>
Date: Sat, 13 Jul 2019 11:13:52 +0000
Subject: [PATCH 2/3] relocatable

---
 rime_config.h |  6 +++---
 rime_engine.c | 25 ++++++++++++++++++-------
 2 files changed, 21 insertions(+), 10 deletions(-)

diff --git a/rime_config.h b/rime_config.h
index 1a83488..514447c 100644
--- a/rime_config.h
+++ b/rime_config.h
@@ -3,8 +3,8 @@
 
 #define RIME_VERSION "1.4.0"
 
-#define IBUS_RIME_INSTALL_PREFIX "/usr"
-#define IBUS_RIME_SHARED_DATA_DIR IBUS_RIME_INSTALL_PREFIX "/share/rime-data"
-#define IBUS_RIME_ICONS_DIR IBUS_RIME_INSTALL_PREFIX "/share/ibus-rime/icons"
+#define IBUS_RIME_INSTALL_PREFIX ""
+#define IBUS_RIME_SHARED_DATA_DIR IBUS_RIME_INSTALL_PREFIX "share/rime-data"
+#define IBUS_RIME_ICONS_DIR IBUS_RIME_INSTALL_PREFIX "share/ibus-rime/icons"
 
 #endif  // RIME_CONFIG_H_
diff --git a/rime_engine.c b/rime_engine.c
index 1348c70..928abce 100644
--- a/rime_engine.c
+++ b/rime_engine.c
@@ -110,42 +110,50 @@ ibus_rime_engine_init (IBusRimeEngine *rime_engine)
   IBusProperty* prop;
   IBusText* label;
   IBusText* tips;
+  gchar *cwd = g_get_current_dir();
+  gchar *icon = g_strdup_printf("%s/%s", cwd, IBUS_RIME_ICONS_DIR "/zh.png");
   label = ibus_text_new_from_static_string("中文");
   tips = ibus_text_new_from_static_string("中 ↔ A");
   prop = ibus_property_new("InputMode",
                            PROP_TYPE_NORMAL,
                            label,
-                           IBUS_RIME_ICONS_DIR "/zh.png",
+                           icon,
                            tips,
                            TRUE,
                            TRUE,
                            PROP_STATE_UNCHECKED,
                            NULL);
   ibus_prop_list_append(rime_engine->props, prop);
+  g_free(icon);
+  icon = g_strdup_printf("%s/%s", cwd, IBUS_RIME_ICONS_DIR "/reload.png");
   label = ibus_text_new_from_static_string("部署");
   tips = ibus_text_new_from_static_string(_("Deploy"));
   prop = ibus_property_new("deploy",
                            PROP_TYPE_NORMAL,
                            label,
-                           IBUS_RIME_ICONS_DIR "/reload.png",
+                           icon,
                            tips,
                            TRUE,
                            TRUE,
                            PROP_STATE_UNCHECKED,
                            NULL);
   ibus_prop_list_append(rime_engine->props, prop);
+  g_free(icon);
+  icon = g_strdup_printf("%s/%s", cwd, IBUS_RIME_ICONS_DIR "/sync.png");
   label = ibus_text_new_from_static_string("同步");
   tips = ibus_text_new_from_static_string(_("Sync data"));
   prop = ibus_property_new("sync",
                            PROP_TYPE_NORMAL,
                            label,
-                           IBUS_RIME_ICONS_DIR "/sync.png",
+                           icon,
                            tips,
                            TRUE,
                            TRUE,
                            PROP_STATE_UNCHECKED,
                            NULL);
   ibus_prop_list_append(rime_engine->props, prop);
+  g_free(icon);
+  g_free(cwd);
 }
 
 static void
@@ -235,19 +243,20 @@ static void ibus_rime_update_status(IBusRimeEngine *rime_engine,
       status && status->schema_id ? g_strdup(status->schema_id) : NULL;
 
   IBusProperty* prop = ibus_prop_list_get(rime_engine->props, 0);
-  const gchar* icon;
+  gchar *cwd = g_get_current_dir();
+  gchar* icon = NULL;
   IBusText* label;
   if (prop) {
     if (!status || status->is_disabled) {
-      icon = IBUS_RIME_ICONS_DIR "/disabled.png";
+      icon = g_strdup_printf("%s/%s", cwd, IBUS_RIME_ICONS_DIR "/disabled.png");
       label = ibus_text_new_from_static_string("維護");
     }
     else if (status->is_ascii_mode) {
-      icon = IBUS_RIME_ICONS_DIR "/abc.png";
+      icon = g_strdup_printf("%s/%s", cwd, IBUS_RIME_ICONS_DIR "/abc.png");
       label = ibus_text_new_from_static_string("Abc");
     }
     else {
-      icon = IBUS_RIME_ICONS_DIR "/zh.png";
+      icon = g_strdup_printf("%s/%s", cwd, IBUS_RIME_ICONS_DIR "/zh.png");
       /* schema_name is ".default" in switcher */
       if (status->schema_name && status->schema_name[0] != '.') {
         label = ibus_text_new_from_string(status->schema_name);
@@ -265,6 +274,8 @@ static void ibus_rime_update_status(IBusRimeEngine *rime_engine,
     ibus_property_set_label(prop, label);
     ibus_engine_update_property((IBusEngine *)rime_engine, prop);
   }
+  g_free(cwd);
+  g_free(icon);
 }
 
 static void ibus_rime_engine_update(IBusRimeEngine *rime_engine)
-- 
2.20.1

