From 7a248b9f43a328d339d146f029fd2f36e2afe046 Mon Sep 17 00:00:00 2001
From: Chunhui He <hchunhui@mail.ustc.edu.cn>
Date: Thu, 18 Jul 2019 05:13:43 +0000
Subject: [PATCH 3/3] my color scheme

---
 rime_engine.c | 33 ++++++++++++++++++++++++---------
 1 file changed, 24 insertions(+), 9 deletions(-)

diff --git a/rime_engine.c b/rime_engine.c
index 0b061e1..54e1e79 100644
--- a/rime_engine.c
+++ b/rime_engine.c
@@ -356,16 +356,19 @@ static void ibus_rime_engine_update(IBusRimeEngine *rime_engine)
     glong cursor_pos =
         g_utf8_strlen(preedit, context.composition.cursor_pos - offset);
     text->attrs = ibus_attr_list_new();
+    glong start = g_utf8_strlen(
+        preedit,context.composition.sel_start - offset);
+    glong end = g_utf8_strlen(preedit, context.composition.sel_end - offset);
+    ibus_attr_list_append(
+        text->attrs,
+        ibus_attr_foreground_new(0x0063c8, 0, start));
+    ibus_attr_list_append(
+        text->attrs,
+        ibus_attr_foreground_new(0xff0084, start, preedit_len));
     if (highlighting) {
-      glong start = g_utf8_strlen(
-          preedit,context.composition.sel_start - offset);
-      glong end = g_utf8_strlen(preedit, context.composition.sel_end - offset);
-      ibus_attr_list_append(
-          text->attrs,
-          ibus_attr_foreground_new(RIME_COLOR_BLACK, start, end));
       ibus_attr_list_append(
           text->attrs,
-          ibus_attr_background_new(RIME_COLOR_LIGHT, start, end));
+          ibus_attr_background_new(0xeeeeec, start, end));
     }
   }
 
@@ -408,20 +411,27 @@ static void ibus_rime_engine_update(IBusRimeEngine *rime_engine)
       gchar* text = context.menu.candidates[i].text;
       gchar* comment = context.menu.candidates[i].comment;
       IBusText *cand_text = NULL;
+      int text_len = g_utf8_strlen(text, -1);
+
       if (comment) {
         gchar* temp = g_strconcat(text, " ", comment, NULL);
         cand_text = ibus_text_new_from_string(temp);
         g_free(temp);
-        int text_len = g_utf8_strlen(text, -1);
         int end_index = ibus_text_get_length(cand_text);
         ibus_text_append_attribute(cand_text,
                                    IBUS_ATTR_TYPE_FOREGROUND,
-                                   RIME_COLOR_DARK,
+                                   0xff0084,
                                    text_len, end_index);
       }
       else {
         cand_text = ibus_text_new_from_string(text);
       }
+
+      ibus_text_append_attribute(cand_text,
+                                 IBUS_ATTR_TYPE_FOREGROUND,
+                                 0x0063c8,
+                                 0, text_len);
+
       ibus_lookup_table_append_candidate(rime_engine->table, cand_text);
       IBusText *label = NULL;
       if (i < context.menu.page_size && has_labels) {
@@ -433,6 +443,11 @@ static void ibus_rime_engine_update(IBusRimeEngine *rime_engine)
       else {
         label = ibus_text_new_from_printf("%d", (i + 1) % 10);
       }
+      int label_len = g_utf8_strlen(ibus_text_get_text(label), -1);
+      ibus_text_append_attribute(label,
+                                 IBUS_ATTR_TYPE_FOREGROUND,
+                                 0x0063c8,
+                                 0, label_len);
       ibus_lookup_table_set_label(rime_engine->table, i, label);
     }
     if (has_page_down) { //show page down except last page
-- 
2.20.1

